src/include/cursor.h

```diff
@@ -586,3 +586,5 @@ struct __wt_cursor_version {
 
 #define WT_CURSOR_RAW_OK \
     (WT_CURSTD_DUMP_HEX | WT_CURSTD_DUMP_PRETTY | WT_CURSTD_DUMP_PRINT | WT_CURSTD_RAW)
+
+#define WT_CURSOR_IS_POSITIONED(cbt) (cbt->ref != NULL && cbt->ref->page != NULL)
```

Comment:
Add a comment here describing why this is correct and what use cases it has for now.

Something like:
```
A positioned cursor must have a page, this is a requirement of the cursor logic within the wiredtiger API. As such 
if the page on the cursor is not null we can safely assume that the cursor is positioned.

This is primarily used by cursor bound checking logic.
```